lib/diagnostic.js

```diff
@@ -5,19 +5,31 @@
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
-const min = function (/* previous, current */) {
+const min = function (previous, current) {
+    const current = previous.slice();
+    let accumulator = current;
 
+    if(typeof current === 'undefined'){
+      accumulator = current.shift();
+    }
+
+    let minimum = array.reduce(min);
 };
 
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let total = array.reduce(sum);`
-const sum = function (/* previous, current */) {
+const sum = function (previous, current)  {
+  for( i = 0; i < 0; i++){
```

Comment:
```i``` needs to be declared with ```let```.  This loop will never run since the initial condition will be ```0 < 0```.